app/experimenter/kinto/tasks.py

```diff
@@ -99,7 +99,14 @@ def nimbus_check_kinto_push_queue():
         if rejected_collection_data:
             rejected_slug = kinto_client.get_rejected_record()
             experiment = NimbusExperiment.objects.get(slug=rejected_slug)
-            experiment.status = NimbusExperiment.Status.DRAFT
+            if (
+                experiment.status == NimbusExperiment.Status.LIVE
+                and experiment.is_end_requested
+            ):
+                experiment.is_end_requested = False
+            else:
+                experiment.status = NimbusExperiment.Status.DRAFT
+
```

Comment:
Probably rare that we'll have an experiment deletion rejected on the kinto side. But, when I tried it, I realized that this code threw the status back into Draft status in Experimenter while the experiment was still Live over in remote settings. This leaves it live but reverts the request to end the experiment. Would be nice if we had some indication of the rejection message, but I don't think we surface rejection messages anywhere yet.

Reply:
The plan is to move the review step into experimenter in https://jira.mozilla.com/browse/EXP-865, so probably makes sense to wait until then to do that

Reply:
Oh good catch 👍

Reply:
> The plan is to move the review step into experimenter in https://jira.mozilla.com/browse/EXP-865, so probably makes sense to wait until then to do that

Yeah, we're going to change the review process in the future - but more immediately, this status change could cause a problem unless we want to delay the end experiment feature